scripts/diagnostics/vulkan/stage_cmp.cpp

```
@@ -0,0 +1,105 @@
+    if (!moss::encode_audio(m, mel, enc, e)) { fprintf(stderr, "enc: %s\n", e.c_str()); return 2; }
+    stat("encoder", enc.data);
+
+    moss::AudioEncoding adap;
+    if (!moss::apply_adapter(m, enc, adap, e)) { fprintf(stderr, "adapter: %s\n", e.c_str()); return 2; }
+    stat("adapter", adap.data);
+
+    auto prompt = moss::build_transcribe_prompt(m.config, 743);
+    moss::InputsEmbeds ie;
+    if (const char* ov = getenv("STAGE_CMP_EMBEDS")) {
+        FILE* g = fopen(ov, "rb");
+        fseek(g, 0, SEEK_END); long gs = ftell(g); fseek(g, 0, SEEK_SET);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)